### PR TITLE
[armadillo]update to 10.6.2

### LINKED ISSUE #19471: [Armadillo ]update to >=10.5.2
Library name: Armadillo

New version number: [10.6.2](http://arma.sourceforge.net/download.html)

We've hit a roadblock creating Python bindings for [our project](https://github.com/stanley-rozell/lds-ctrl-est) since I need CARMA to convert between Armadillo and NumPy, yet CARMA [requires at least Armadillo 10.5.2](https://carma.readthedocs.io/en/latest/introduction.html#requirements).

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #19471 

Update armadillo to the latest version 10.6.2

Note: no feature need to tested

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@kjohnsen ,I tested the usage of armadillo locally and the test was successful. If you have a usage problem, you can reopen an issue and we will resolve it for you as soon as possible

File: ports/armadillo/portfile.cmake [hunk 1]
```diff
@@ -3,46 +3,43 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
-    SOURCE_PATH ${SOURCE_PATH}
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     DISABLE_PARALLEL_CONFIGURE
-    PREFER_NINJA
     OPTIONS
         -DDETECT_HDF5=false
         -DCMAKE_DISABLE_FIND_PACKAGE_SuperLU=ON
-        -DCMAKE_DISABLE_FIND_PACKAGE_ACML=ON
-        -DCMAKE_DISABLE_FIND_PACKAGE_ACMLMP=ON
```
Comment (JonLiu1993) [inline ports/armadillo/portfile.cmake:26]: These options are not used in the latest version and cannot be found in the source

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the version update!

**Pull request merged**